### DISCUSSION TITLE
Periastron Overflowing When Adopting Solution

### DISCUSSION THREAD

**Discussion by nbungart1999** [phoebe-project/phoebe2#465] [Questions]
I am trying to adopt the solution from the ebai estimator and I am getting the warning that the primary is overflowing at the periastron. I know that this means the value is outside the range and it does say that this will cause an error when I do run_compute. My question is should I manually change the value but then I'm not truly adopting the solution or is there some other method to fix this?
![image](https://user-images.githubusercontent.com/83960306/121559981-16ac0c80-c9e5-11eb-9dff-c29c9275443c.png)

**Comment by kecnry** [comment]
This may just be an intermediate warning (raised while some parameters are adopted before others).  You can always call [b.run_checks()](http://phoebe-project.org/docs/2.3/api/phoebe.frontend.bundle.Bundle.run_checks) after to see if they have cleared or are still failing.  

To disable these intermediate logger checks, you can call [phoebe.interactive_checks_off()](http://phoebe-project.org/docs/2.3/api/phoebe.interactive_checks_off) (which is planned to be the new default in the next release as this seems to trip a lot of people, and is especially confusing in notebook with multiple lines executing in one cell).

If your system is still overflowing, then you'll have to adjust some parameters so it is not or choose to only adopt a subset of the proposed parameters.  Ebai is not _always_ the best at giving a good estimate (it depends on the system), so I would always suggest checking to make sure the values are reasonable.  If lc geometry works fine for your system, then I would even suggest sticking to that for now and just optimizing for the temperature ratio and requivs instead.

**Reply by nbungart1999** [reply]
Ok, so after running the checks I do have 1 failing item, I just tried to also manually set the requiv of the primary but it is constrained because of how I set some of the other estimators.

**Reply by kecnry** [reply]
Ok, so then you'll need to change some combination of the parameters that affect requiv or requiv_max (there are many of them) such that it no longer overflows.  You can access the equations (and parameters) that go into these with something like:

```
print(b.filter(qualifier=['requiv', 'requiv_max'], component='primary', context='constraint'))
```

Alternatively, if you know that your system is fully-filling the roche lobe (semi-detached) at periastron, you can force it to [always be at the critical value](http://phoebe-project.org/docs/2.3/tutorials/requiv_crit_semidetached) and won't have to worry about it overflowing.
